### DISCUSSION TITLE
How To Create A IPSec Tunnel With A Specific WAN Interface?

### DISCUSSION THREAD

**Discussion by mrsagarjain1** [strongswan/strongswan#817] [Q&A]
Hi, 
I am using OpenWrt on my router I have two WAN connections on my router. Both WAN interfaces have dynamic public IPs. I am able to create a single IPSec tunnel. The tunnel was created through the WAN interface which has the lowest metric in the routing table.
I want to ask you that is there any way to create the IPSec tunnel through a specific WAN interface?
Thanks.

![image](https://user-images.githubusercontent.com/83968938/145983703-faaafeba-463b-4a82-880f-e2f35178ee7e.png)

**Comment by tobiasbrunner** [comment]
Depends mainly on the routes (i.e. try to control it that way, note that MOBIKE can change the path/local address dynamically on route changes), but also the local address that's configured for a connection (I guess not an option if they are dynamic). You could also ignore specific interfaces in the config, but that's a global option that requires a restart to change.

**Reply by mrsagarjain1** [reply]
@tobiasbrunner 
So technically we cannot create two tunnels with two different WANs which have dynamic public IPs. Am I correct?

**Reply by tobiasbrunner** [reply]
If you can connect to two different remote addresses, you could install routes via different interfaces. Or you might use different local host names that resolve to the two dynamic IPs once they are known, which could then be configured for two connections (note that strongSwan currently won't abort connecting if the source address is unresolved/unknown, it will just try to send packets from whatever address is possible). Another option might be to use two private addresses that you NAT each on a different interface (not ideal if there is otherwise no NAT). The question is whether it's actually necessary to have two connections, because MOBIKE can switch a connection from one interface to the other depending on address/interface/route changes.
